### PR TITLE
Improve the readbility of the code by removing the else branch

### PR DESCRIPTION
**Pull request by sylvestre**
after a return, break or continue.

Found by clang-tidy

### CODE REVIEW

File: lib/curl_addrinfo.c [hunk 1]
```diff
@@ -469,12 +469,10 @@ Curl_addrinfo *Curl_str2addr(char *address, int port)
     /* This is a dotted IP address 123.123.123.123-style */
     return Curl_ip2addr(AF_INET, &in, address, port);
 #ifdef ENABLE_IPV6
-  else {
-    struct in6_addr in6;
-    if(Curl_inet_pton(AF_INET6, address, &in6) > 0)
-      /* This is a dotted IPv6 address ::1-style */
-      return Curl_ip2addr(AF_INET6, &in6, address, port);
-  }
+  struct in6_addr in6;
```
Comment (bagder) [inline lib/curl_addrinfo.c]: This is not C89 compliant as it mixes variable declarations and code.

File: lib/ftp.c [hunk 2]
```diff
@@ -1679,26 +1678,24 @@ static CURLcode ftp_state_ul_setup(struct connectdata *conn,
...
-          passed += actuallyread;
-          if((actuallyread == 0) || (actuallyread > readthisamountnow)) {
-            /* this checks for greater-than only to make sure that the
-               CURL_READFUNC_ABORT return code still aborts */
-            failf(data, "Failed to read data");
-            return CURLE_FTP_COULDNT_USE_REST;
-          }
-        } while(passed < data->state.resume_from);
-      }
+      curl_off_t passed=0;
```
Comment (bagder) [inline lib/ftp.c]: This as well.

File: lib/http.c [hunk 3]
```diff
@@ -2152,27 +2151,25 @@ CURLcode Curl_http(struct connectdata *conn, bool *done)
...
-            if((actuallyread == 0) || (actuallyread > readthisamountnow)) {
-              /* this checks for greater-than only to make sure that the
-                 CURL_READFUNC_ABORT return code still aborts */
-              failf(data, "Could only read %" CURL_FORMAT_CURL_OFF_T
-                    " bytes from the input", passed);
-              return CURLE_READ_ERROR;
-            }
-          } while(passed < data->state.resume_from);
-        }
+        curl_off_t passed=0;
```
Comment (bagder) [inline lib/http.c]: and here

**Comment by bagder** [comment]
A very basic merge conflict to fix but more importantly more C89 nits:

~~~
transfer.c:130:5: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
     struct SingleRequest *k = &data->req;
     ^~~~~~
ssh.c:1123:9: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
         char *err_msg;
         ^~~~
ssh.c:1837:11: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
           curl_off_t passed=0;
           ^~~~~~~~~~
ssh.c:2428:9: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
         int ssh_err;
         ^~~
ssh.c:2496:9: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
         int ssh_err;
         ^~~
ssh.c:2831:5: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
     struct timeval now = Curl_tvnow();
     ^~~~~~

**Comment by bagder** [comment]
The test 1307 failure in Travis seems to be caused by this PR, it is an otherwise stable test. Oh, and please squash/rebase if you can, to make it easier for me!

**Comment by sylvestre** [comment]
@bagder problem fixed \o/
ready to be merged from my pov

**Comment by bagder** [comment]
Thanks a lot!

**Pull request closed without merging**